### ISSUE TITLE
test failed in CI: ls-apis

### ISSUE THREAD

**Issue by smklein** [oxidecomputer/omicron#6902]
This test failed on a CI run on  https://github.com/oxidecomputer/omicron/pull/6881

https://github.com/oxidecomputer/omicron/pull/6881/checks?check_run_id=31707385474

Log showing the specific test failure:

https://buildomat.eng.oxide.computer/wg/0/details/01JAEH7HTJQS6CJNX1ZV5VY03E/xoPOQFjBiuMJmGrEoSVqo5qACfemYSsHmRVoN9ObrIV6V13M/01JAEH7ZCEDZE5CQFW7CMP2GR3

Excerpt from the log showing the failure:

```
4433	2024-10-18T01:44:53.308Z	     Running `target/debug/ls-apis apis`
4434	2024-10-18T01:44:53.476Z	loading metadata for workspace omicron from current workspace
4435	2024-10-18T01:44:55.320Z	loading metadata for workspace maghemite from /home/build/.cargo/git/checkouts/maghemite-de41bdd6c14939ab/056283e/Cargo.toml
4436	2024-10-18T01:44:55.323Z	loading metadata for workspace propolis from /home/build/.cargo/git/checkouts/propolis-12517f89d3d9f483/11371b0/Cargo.toml
4437	2024-10-18T01:44:55.326Z	loading metadata for workspace crucible from /home/build/.cargo/git/checkouts/crucible-f3b5bdecdc6486d6/2b88ab8/Cargo.toml
4438	2024-10-18T01:45:03.716Z	Error: loading metadata
4439	2024-10-18T01:45:03.716Z	
4440	2024-10-18T01:45:03.716Z	Caused by:
4441	2024-10-18T01:45:03.716Z	    `cargo metadata` exited with an error:     Updating git repository `https://www.github.com/oxidecomputer/oxnet`
4442	2024-10-18T01:45:03.716Z	    warning: redirecting to https://github.com/oxidecomputer/oxnet.git/
4443	2024-10-18T01:45:03.716Z	    From https://www.github.com/oxidecomputer/oxnet
4444	2024-10-18T01:45:03.716Z	     * [new branch]      ci         -> origin/ci
4445	2024-10-18T01:45:03.716Z	     * [new branch]      dependabot/cargo/regress-0.10.1 -> origin/dependabot/cargo/regress-0.10.1
4446	2024-10-18T01:45:03.716Z	     * [new branch]      dependabot/cargo/serde-1.0.210 -> origin/dependabot/cargo/serde-1.0.210
4447	2024-10-18T01:45:03.717Z	     * [new branch]      dependabot/cargo/serde_json-1.0.128 -> origin/dependabot/cargo/serde_json-1.0.128
4448	2024-10-18T01:45:03.717Z	     * [new branch]      main       -> origin/main
4449	2024-10-18T01:45:03.717Z	     * [new ref]                    -> origin/HEAD
4450	2024-10-18T01:45:03.717Z	        Updating git repository `https://github.com/oxidecomputer/dendrite`
4451	2024-10-18T01:45:03.717Z	    fatal: could not read Username for 'https://github.com': No such device or address
4452	2024-10-18T01:45:03.717Z	    error: failed to get `common` as a dependency of package `ddm v0.1.0 (/home/build/.cargo/git/checkouts/maghemite-de41bdd6c14939ab/056283e/ddm)`
4453	2024-10-18T01:45:03.717Z	
4454	2024-10-18T01:45:03.717Z	    Caused by:
4455	2024-10-18T01:45:03.717Z	      failed to load source for dependency `common`
4456	2024-10-18T01:45:03.717Z	
4457	2024-10-18T01:45:03.717Z	    Caused by:
4458	2024-10-18T01:45:03.717Z	      Unable to update https://github.com/oxidecomputer/dendrite?branch=main#4cdc7d7e
4459	2024-10-18T01:45:03.717Z	
4460	2024-10-18T01:45:03.717Z	    Caused by:
4461	2024-10-18T01:45:03.717Z	      failed to clone into: /home/build/.cargo/git/db/dendrite-627b239a911c6241
4462	2024-10-18T01:45:03.717Z	
4463	2024-10-18T01:45:03.717Z	    Caused by:
4464	2024-10-18T01:45:03.717Z	      process didn't exit successfully: `git fetch --force --update-head-ok 'https://github.com/oxidecomputer/dendrite' '+4cdc7d7e18a25833843686d76b95a9ca4a5dfab0:refs/commit/4cdc7d7e18a25833843686d76b95a9ca4a5dfab0'` (exit status: 128)
4465	2024-10-18T01:45:03.820Z	
```

**Comment by smklein** [comment]
@davepacheco this looks like the ls-apis command depends on external services and might fail - seems like it might benefit from a retry loop?

**Comment by davepacheco** [comment]
The problem here is that this was running for more than an hour (just barely) and the GitHub access token expired.  I think this is a dup of #6839, which I think @iliana was going to fix in #6840.
